src/internal/github/rss.go

```diff
@@ -42,37 +40,15 @@ func extractTag(item *gofeed.Item) (string, error) {
 	return matches[pattern.SubexpIndex("Version")], nil
 }
 
-func getReleaseRssFeed(releasesRssUrl string) (feed *gofeed.Feed, err error) {
-	client := httpInternal.GetHttpRetryClient()
-
-	req, err := http.NewRequestWithContext(context.Background(), "GET", releasesRssUrl, nil)
-	if err != nil {
-		return nil, err
-	}
-
-	// TODO Commonize?
-	token := os.Getenv("GH_TOKEN")
-
-	req.Header.Set("User-Agent", "OpenTofu Registry/1.0")
-	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token))
-
-	resp, err := client.Do(req)
-
+func (c Client) getReleaseRssFeed(releasesRssUrl string) (*gofeed.Feed, error) {
+	resp, err := c.httpClient.Get(releasesRssUrl)
 	if err != nil {
-		return nil, err
-	}
-
-	if resp != nil {
-		defer func() {
-			ce := resp.Body.Close()
-			if ce != nil {
-				err = ce
-			}
-		}()
+		return nil, fmt.Errorf("%s: %w", releasesRssUrl, err)
 	}
+	defer resp.Body.Close()
 
-	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
-		return nil, fmt.Errorf("got error %d", resp.StatusCode)
+	if resp.StatusCode != http.StatusOK {
```

Comment:
**(idea)** WDYT about covering 429 with a backoff/retry pattern?

Reply:
That's a good thought, though it would be neat to do that inside of the backing httpClient transparently.  There's some commented out code in github.NewClient that had a similar idea.  Github's API theoretically has some good response headers that can tell us if we are hitting it too hard.

I think this is an ongoing discussion and might be worth creating an issue in the repo?